### PR TITLE
Dgad 422 - Review and merge

### PR DESCRIPTION
**Pull request by starsinmypockets**
@boutell Review and merge

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -224,6 +224,13 @@ And, an equivalent configuration for Azure:
       disabledFileKey: 'a random string of your choosing',
       // Always required for Azure
       tempPath: __dirname + '/temp',
+      // by default we gzip encode EVERYTHING except for a short list of excpetions, found in defaultGzipBlacklist.js
```
Comment (boutell) [inline README.md]: typo

File: defaultGzipBlacklist.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// By default, the following file types will not be gzipped
+module.exports = ['jpg', 'png', 'zip', 'gzip', 'png', 'xls', 'docx', 'gif'];
```
Comment (boutell) [inline defaultGzipBlacklist.js]: xlsx. Classic xls is not a zip file. Others to add: xltx, dotx, pptx, ppsx, potx, sldx, gz.

File: lib/storage/azure.js [hunk 3]
```diff
@@ -5,8 +5,11 @@ var fs = require('fs');
 var zlib = require('zlib');
 var async = require('async');
 var utils = require('../utils.js');
+var defaultGzipBlacklist = require('../../defaultGzipBlacklist');
 var verbose = false;
+var _ = require('lodash');
 
+var gzip = zlib.createGzip();
```
Comment (boutell) [inline lib/storage/azure.js]: Whoa, you rolled back my fix to the bug where a single instance of the zlib stream was reused; what else did you roll back when you merged?

**Comment by starsinmypockets** [comment]
@boutell I merged your previous work - let me know if this looks legit to you. I need to refactor tests (don't work now with non-gzipped formats)

File: test/azure.js [hunk 4]
```diff
@@ -62,6 +63,41 @@ describe('UploadFS Azure', function() {
...
+  });
+
+  it('getGzipBlacklist should be able to add a type to the blacklist based on user settings', done => {
+    const types = uploadfs._storage.getGzipBlacklist({ 'foo': false });
+    assert(Array.isArray(types), 'gzip blacklist array is an array');
+    assert(types && types.indexOf('foo' >= 0));
+    done();
+  });
+
+  it('getGzipBlacklist should be pretend to remove an item from the blacklist that was never on it based on user options', done => {
```
Comment (boutell) [inline test/azure.js]: Fix grammar here, not sure what you're saying

**Pull request merged**